src/odemis/dataio/tiff.py

```diff
@@ -2722,3 +2742,188 @@ def _convert_thermo_fisher_to_odemis_metadata(metadata: str) -> dict:
         logging.warning(f"Failed to parse ThermoFisher metadata: {model.MD_DET_TYPE} - {e}")
 
     return md
+
+### Tescan Metadata parsing
+def _parse_tescan_metadata(metadata: bytes) -> str:
+    """
+    Parses TESCAN metadata bytes to str
+    :param metadata: metadata bytes (TESCAN bytes format)
+    :return: (str) metadata (TESCAN ini format)
+    """
+    # NOTE:
+    # METADATA FORMAT
+    # JPEG2000 metadata
+    #   4 bytes: length of the next chunk (n)
+    #   2 bytes: header of the next chunk (type)
+    #   n bytes: chunk data
+    # MAIN metadata
+    #   4 bytes: length of the next chunk (n)
+    #   2 bytes: header of the next chunk (type)
+    #   n - 3 bytes: valid chunk data
+    #   3 bytes: unsure?
+    # SEM/FIB metadata
+    #   4 bytes: length of the next chunk (n)
+    #   2 bytes: header of the next chunk (type)
+    #   n - 3 bytes: valid chunk data
+    #   3 bytes: unsure?
+
+    # constants
+    _LENGTH_BYTES = 4
+    _HEADER_BYTES = 2
+    _END_BYTES = 3
+
+    # valid metadata str
+    vmd = ""
+
+    # header type map
+    _header_map ={
+        2: "[MAIN]",
+        3: "[SEM]",
+        5: "[FIB]"
+    }
+
+    # skip jpeg2000 metadata
+    next_chunk_len = int.from_bytes(metadata[:4], byteorder="little", signed=False)
+    
+    # loop through the two chunks, extract the valid data
+    for i in range(2):
+        # skip to next chunk
+        metadata = metadata[4+next_chunk_len:] # skip to next chunk
+        
+        # read the length of the next chunk (first 4 bytes)
+        next_chunk_len = int.from_bytes(metadata[:_LENGTH_BYTES], 
+                                            byteorder="little", signed=False)
+        
+        # read the type of the chunk (next 2 bytes)
+        next_chunk_header = int.from_bytes(metadata[_LENGTH_BYTES:_LENGTH_BYTES+_HEADER_BYTES], 
+                                            byteorder="little", signed=False)
+        
+        # read the chunk data as string (next n-3 bytes)
+        chunk_start = _LENGTH_BYTES+_HEADER_BYTES
+        chunk_end = chunk_start + next_chunk_len - _END_BYTES
+        next_chunk = metadata[chunk_start:chunk_end].decode("latin1")
+
+        # read the final 3 bytes (unused)
+        # NOTE: 3 bytes at the end of each chunk, not sure what for?
+        # standard values were 768, 512, 1208, so maybe something to do with display?        
+        next_chunk_final = metadata[chunk_end:chunk_end+_END_BYTES]
+        
+        # add the chunk data to the metadata string
```

Comment:
Skip if `next_chunk_header` not in `header_map` (to handle other metadata gracefully)